include/swift/AST/Decl.h

```diff
@@ -5164,6 +5164,7 @@ class ParamDecl : public VarDecl {
     PointerUnion<Expr *, VarDecl *> DefaultArg;
     Initializer *InitContext = nullptr;
     StringRef StringRepresentation;
+    CaptureInfo Captures;
```

Comment:
This is, uh, kind of expensive to put in every default argument, no?

Reply:
I guess, but I don't think there are that many default arguments. We can shrink it down to the size of a single ArrayRef with a bit of work though.

Reply:
Maybe you'll have successfully nerd-sniped me into it after all!

Reply:
Hah, you're more than welcome to. I have another commit I'm going to tack onto this PR that implements default argument captures for local functions -- it wasn't too hard after all. Might want to wait for that one to avoid too many merge conflicts